From c53d6ce32d2e4420fcac3f2d68547c3b07e4fce6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Thu, 4 Sep 2014 14:02:55 +0200
Subject: [PATCH] Accept DateTime::TimeZone::Tzfile object in place of
 DateTime::TimeZone
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

After fixing <https://bugzilla.redhat.com/show_bug.cgi?id=1135981> by
returning DateTime::TimeZone::Tzfile, we need to allow this class in
this library.

https://bugzilla.redhat.com/show_bug.cgi?id=1138185
Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 lib/MooseX/Types/DateTime/ButMaintained.pm | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/lib/MooseX/Types/DateTime/ButMaintained.pm b/lib/MooseX/Types/DateTime/ButMaintained.pm
index 458d9a5..44bf3aa 100644
--- a/lib/MooseX/Types/DateTime/ButMaintained.pm
+++ b/lib/MooseX/Types/DateTime/ButMaintained.pm
@@ -10,7 +10,7 @@ use DateTime::Locale ();
 use DateTime::TimeZone ();
 use Olson::Abbreviations 0.03 qw();
 
-use MooseX::Types::Moose 0.30 qw/Num HashRef Str/;
+use MooseX::Types::Moose 0.30 qw/Num HashRef Str Object/;
 
 use MooseX::Types 0.30 -declare => [qw( DateTime Duration TimeZone Locale Now )];
 
@@ -23,7 +23,9 @@ class_type "DateTime::Locale::root" => { name => "DateTime::Locale" };
 
 subtype DateTime, as 'DateTime';
 subtype Duration, as 'DateTime::Duration';
-subtype TimeZone, as 'DateTime::TimeZone';
+subtype TimeZone, as Object, where {
+    $_->isa('DateTime::TimeZone') || $_->isa('DateTime::TimeZone::Tzfile')
+};
 subtype Locale,   as 'DateTime::Locale';
 
 subtype( Now, as Str, where { $_ eq 'now' },
-- 
1.9.3

